app/waf/test/acceptance/adc.controller.acceptance.ts

```diff
@@ -193,14 +203,30 @@ describe('AdcController', () => {
       ],
     });
 
-    const response = await client
+    this.timeout(5000);
+
+    function sleep(ms: number) {
+      return new Promise(resolve => setTimeout(resolve, ms));
+    }
+
+    let response = await client
       .post(prefix + '/adcs')
       .set('X-Auth-Token', ExpectedData.userToken)
       .set('tenant-id', ExpectedData.tenantId)
       .send(adc)
       .expect(200);
 
     expect(response.body.adc).to.containDeep(toJSON(adc));
+
+    await sleep(3000);
```

Comment:
This test case is attempt to simulating the real scenario. An api call comes, then our code attempt to trust a device. The adc state is NONE at the beginning, and becomes TRUSTING after invoking ASG api, and becoming TRUSTED eventually after ASG replies that trusted device is active. That is asynchronous, so the client need another api call to get the TRUSTED state response. Your checkAndWait() interval is 1 second, so I wait 3 seconds to ensure I can get TRUSTED state.

Reply:
How much value for you to catch the NONE state??
You can leave it here, I'll eventually remove it.

Reply:
I was not attempting to catch NONE state. I was attempting to catch TRUSTED state. I need to wait a few seconds, since the state change is done in background.

Reply:
Could you please refer to `await checkAndWait(checkStatus, 5, [], 50).then(() => {`

even little better isn't it??

Reply:
If we can change checkAndWait() default interval during UT, that should be a solution.

Reply:
yes, it can be changed as shown in the example: 50 ms. I think 10 ms or 5 ms may be better.

Reply:
I post a request to set the interval in UT. Please look at it first. https://github.com/F5Networks/f5-adcaas-openstack/pull/148